File: java/src/main/java/variable/Keyword.java
package variable;

import java.util.Arrays;
import java.util.List;

public class Keyword {

    public static void main(String[] args) {
        long number1 = 10L; // (1) 가변
        final long number2 = 10L; // (2) 불변

        number1 = 15L;
//        number2 = 20L; 컴파일 에러

        Long number3 = 1_000L; // (3)
        Person person = new Person("김영재"); // (4)

        final List<Integer> numbers = Arrays.asList(1, 2);
        // 컬렉션 자체는 불변이지만, 컬렉션 내부의 요소는 가변일 수 있음
//        numbers = List.of(1, 2, 3); 컴파일 에러
        numbers.add(3);
        System.out.println(numbers);

//        int n = null; 원시 타입 변수는 null을 가질 수 없음
        Integer n = null; // 참조 타입 변수는 null을 가질 수 있음
    }

}

File: kotlin/src/main/kotlin/section1/variable/Keyword.kt
package section1.variable

val PI = 3.141592 // can declare variables at the top level

fun main() {

    var number1 = 10L
    number1 = 5L
    val number2 = 20L
//    number2 = 15L -> 재할당 시 컴파일 에러

    var number3: Long
    var number4 = 30L
//    var number5 = number3 + number4
//    println(number5) -> 초기값 없이 사용 시 컴파일 에러

    val number6: Long
    number6 = 15L // val 최초 한 번만 할당 가능
    println(number6)

    val numbers = listOf(1, 2)
//    numbers = listOf(3, 4) 컴파일 에러

}

fun primitiveType() {

    var number1: Long = 10L // long은 존재하지 않음
}

fun nullType() {

    var number1: Long? = null
//    var number2: Long = null -> null 불가능, 컴파일 에러

}

fun instance() {

    var person = Person("김영재")
}
